qemu/tests/watchdog.py

```diff
@@ -87,9 +87,16 @@ def _action_check(test, session, watchdog_action):
         response_timeout = int(params.get("response_timeout", '240'))
         error_context.context("Check whether or not watchdog action '%s' took"
                               " effect" % watchdog_action, logging.info)
+        if watchdog_action == "inject-nmi":
+            if not utils_misc.wait_for(lambda: "NMI received" in session.get_output(),
+                                       response_timeout, 0, 1):
```

Comment:
Seems that this check only works for few platforms:
```
$ grep "NMI received" -nr
arch/x86/kernel/nmi.c:283:	pr_emerg("Uhhuh. NMI received for unknown reason %02x on CPU %d.\n",
arch/alpha/kernel/sys_jensen.c:102:		printk("Whee.. NMI received. Probable hardware error\n");
arch/sparc/kernel/sun4m_irq.c:310:	printk(KERN_ERR "Aieee: sun4m NMI received!\n");
```

Reply:
Yes, I just checked the results on  power8, they are different, the guest will be rebooted when watchdog is triggered. @qunfang hi, could you confirm the case on powerpc and s390x?

Reply:
Yanhui,  @wangxianxian is working on case confirmation and patch review.

Reply:
@wangxianxian Any update on this? Thanks.

Reply:
On powerpc, after watchdog is triggered, guest will reboot, while on s390x, guest poweroff.
on powerpc:
(1/1) Host_RHEL.RHEL_ALT.m7.u6.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.ALT.7.6.ppc64le.io-github-autotest-qemu.watchdog.watchdog_action.inject_nmi.i6300esb: FAIL: Guest didn't receive debug info with 'NMI received',so action 'inject-nmi' didn't take effect. (322.50 s)

on s390x:
(1/1) Host_RHEL.RHEL_ALT.m7.u6.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.s390x.io-github-autotest-qemu.watchdog.watchdog_test_suit.diag288.s390-virtio: FAIL: Watchdog test suit doesn't run successfully. (35.21 s)
 
I have tried this case manually on powerpc and x390x, the result are as following:
on powerpc:
{"timestamp": {"seconds": 1539769227, "microseconds": 405454}, "event": "WATCHDOG", "data": {"action": "inject-nmi"}}
{"timestamp": {"seconds": 1539769232, "microseconds": 249694}, "event": "RESET", "data": {"guest": true}}
{"timestamp": {"seconds": 1539769272, "microseconds": 537230}, "event": "RTC_CHANGE", "data": {"offset": 1}}

on x390x:
{"timestamp": {"seconds": 1539769505, "microseconds": 899286}, "event": "WATCHDOG", "data": {"action": "inject-nmi"}}
{"timestamp": {"seconds": 1539769505, "microseconds": 899938}, "event": "GUEST_PANICKED", "data": {"action": "pause", "info": {"core": 0, "psw-addr": 1123826, "reason": "disabled-wait", "psw-mask": 562956395872256, "type": "s390"}}}
{"timestamp": {"seconds": 1539769505, "microseconds": 899990}, "event": "GUEST_PANICKED", "data": {"action": "poweroff", "info": {"core": 0, "psw-addr": 1123826, "reason": "disabled-wait", "psw-mask": 562956395872256, "type": "s390"}}}
{"timestamp": {"seconds": 1539769505, "microseconds": 900120}, "event": "SHUTDOWN", "data": {"guest": true}}

Reply:
Because the results of ppc and s390x are different from x86_64, guest will just keep running for x86_64, guest will reboot for ppc, and guest will power off for x390x. @qunfang @wangxianxian Could you help confirm whether they are correct and expected results? If so, pls update the case correspondingly, I will auto it for ppc and s390x according your results.